api/osb/osb_controller.go

```diff
@@ -53,48 +52,41 @@ func (c *Controller) handler(request *web.Request) (*web.Response, error) {
 	}
 	target, _ := url.Parse(broker.BrokerURL)
 
-	reverseProxy := httputil.ReverseProxy{
-		Director: func(req *http.Request) {
-			req.Host = target.Host
-		},
-	}
-
 	username, password := broker.Credentials.Basic.Username, broker.Credentials.Basic.Password
-
-	modifiedRequest := request.Request.WithContext(request.Context())
 	plaintextPassword, err := c.Encrypter.Decrypt([]byte(password))
 	if err != nil {
 		return nil, err
 	}
-	modifiedRequest.SetBasicAuth(username, string(plaintextPassword))
-	modifiedRequest.URL.Scheme = target.Scheme
-	modifiedRequest.URL.Host = target.Host
-	modifiedRequest.Body = ioutil.NopCloser(bytes.NewReader(request.Body))
-	modifiedRequest.ContentLength = int64(len(request.Body))
+
+	proxier := proxy.NewReverseProxy(proxy.Options{
+		Transport: http.DefaultTransport,
```

Comment:
Should not use default Transport here. Can't think of better way at the moment. Perhaps configure it outside of the controller and pass it to the controller?